services/motion/builtin/builtin.go

```diff
@@ -217,66 +219,70 @@ func (ms *builtIn) MoveOnGlobe(
 ) (bool, error) {
 	operation.CancelOtherWithLabel(ctx, builtinOpLabel)
 
-	// get the localizer from the componentName
-	localizer, ok := ms.localizers[movementSensorName]
-	if !ok {
-		return false, resource.DependencyNotFoundError(movementSensorName)
-	}
-
-	currentPosition, dstPIF, err := ms.getRelativePositionAndDestination(ctx, localizer, componentName, movementSensorName, *destination)
-	if err != nil {
-		return false, err
-	}
-
-	plan, kb, err := ms.planMoveOnGlobe(ctx,
+	plan, kb, err := ms.planMoveOnGlobe(
+		ctx,
 		componentName,
-		currentPosition,
-		dstPIF,
-		localizer,
+		destination,
+		movementSensorName,
 		obstacles,
 		linearVelocityMillisPerSec,
 		angularVelocityDegsPerSec,
 		extra,
 	)
 	if err != nil {
-		return false, err
+		return false, fmt.Errorf("error making plan for MoveOnMap: %v", err)
 	}
 
 	// execute the plan
-	for _, step := range plan {
-		for _, inputs := range step {
-			if ctx.Err() != nil {
-				return false, ctx.Err()
-			}
-			if len(inputs) == 0 {
-				continue
-			}
-			if err := kb.GoToInputs(ctx, inputs); err != nil {
-				return false, err
-			}
+	for i := 1; i < len(plan); i++ {
+		ms.logger.Info(plan[i])
+		if err := kb.GoToInputs(ctx, plan[i]); err != nil {
+			return false, err
 		}
 	}
-
 	return true, nil
 }
 
-// planMoveOnGlobe returns the plan for MoveOnGlobe to execute.
+// planMoveOnGlobe returns the plan for MoveOnGlobe to execute
 func (ms *builtIn) planMoveOnGlobe(
 	ctx context.Context,
 	componentName resource.Name,
-	currentPosition r3.Vector,
-	dstPIF *referenceframe.PoseInFrame,
-	localizer motion.Localizer,
+	destination *geo.Point,
+	movementSensorName resource.Name,
 	obstacles []*spatialmath.GeoObstacle,
 	linearVelocityMillisPerSec float64,
 	angularVelocityDegsPerSec float64,
 	extra map[string]interface{},
-) ([]map[string][]referenceframe.Input, kinematicbase.KinematicBase, error) {
-	// create a new empty framesystem which we add our base to
-	fs := referenceframe.NewEmptyFrameSystem("")
+) ([][]referenceframe.Input, kinematicbase.KinematicBase, error) {
+	// build the localizer from the movement sensor
+	movementSensor, ok := ms.movementSensors[movementSensorName]
+	if !ok {
+		return nil, nil, resource.DependencyNotFoundError(movementSensorName)
+	}
+	origin, _, err := movementSensor.Position(ctx, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// add an offset between the movement sensor and the base if it is applicable
+	baseOrigin := referenceframe.NewPoseInFrame(componentName.ShortName(), spatialmath.NewZeroPose())
```

Comment:
Nothing to change for this PR, but I think the string-based (or at least short-name-based) keying for the FS service should be deprecated and replaced with resource names, with a slowly growing urgency.

Reply:
Yeah hopefully we'll have time to do that some time in the near future 😅